### ISSUE TITLE
Request: Native prefers-reduced-motion support

### ISSUE THREAD

**Issue by aardrian** [KenjiBaheux/portals#33]
Safari and now Firefox support user preferences to limit motion and animations through the `prefers-reduced-motion` media query. Chrome, Edge, Opera, etc do not.

Instead of waiting for browser support for `prefers-reduced-motion` media queries, I request a requirement in this spec that any browser implementation must honor user preferences to limit motion and animations.

Further, I would like this spec to ensure its code syntax can honor (work with) media query syntax.

I know from experience (which includes structured user testing) with Internet Explorer's 1999-era page transitions feature, presentations with animated transitions (both in auditoriums and in self-directed training), some native apps, and the iOS 7 UI, that animated transitions can cause problems for users with vestibular disorders. Let's encode support for those users from the start.

**Comment by aardrian** [comment]
- The explainer shows a series of images that imply the new page animates into place.
- The text also says "seamless transition between an inset state and a navigated state", which implies an animation.
- The animated GIF on the [Dev Summit recap](https://blog.chromium.org/2018/11/chrome-dev-summit-2018-day-2-ready-for.html) shows new pages fade into place (which is an animation, though non-directional).
- In the Key Scenarios page, terms like "expanding the inset" describe an animation.

While the spec so far does not specify discrete animations or motions, it does extensively talk about transitions. While it may not in the future dictate the nature of those transitions, the output is necessarily a form of animation.

**Comment by jeremyroman** [comment]
We anticipate authors using existing animation tools to create smooth transitions between pages, but as we currently see it, the portals implementation itself would do a hard cut from one browsing context to another.

If Chromium browsers (incl. Chrome and Opera) aren't doing the right thing for users who prefer reduced motion, I would suggest finding or filing an appropriate bug on our bug tracker, https://crbug.com. I believe Microsoft also has a bug tracker where you can request that support from them.

**Comment by domenic** [comment]
There may be an action item here to update the spec and/or explainer to make it clearer that any animations would be supplied by authors, though, to prevent further confusion of this sort.

**Comment by jeremyroman** [comment]
Possibly. I'm not exactly sure how to clarify it without having to make specific reference to this case (which is important, though it's one of a number of things an application author may have to take into account). I've tried fiddling with the wording, but haven't found a way to call that out that doesn't the text difficult to scan or make it sound like we think that use case doesn't matter (it does -- it's just not part of this specification). Happy to look if you have better wording.

For context:

The specification actually doesn't currently use the word "transition" or "animation" at all (if we did discuss it in non-normative text, then clarifying that it isn't specified makes sense). 

The explainer already has this text as a non-goal:

>  - Specifying transitions or defining APIs related to transitions: the proposal assumes that existing animations and DOM manipulations are enough to create compelling user experiences. We might discover important gaps but these should be addressed via separate efforts to avoid overly specific solutions.

If Chromium does get support for [prefers-reduced-motion](https://bugs.chromium.org/p/chromium/issues/detail?id=722548), it seems that this should fit fairly well with our approach, and I think the same will hold true if Chrome ever does some intervention on the user's behalf to reduce motion.
